src/python/pants/backend/python/macros/python_artifact.py

```diff
@@ -78,6 +80,52 @@ def binaries(self):
     def __str__(self) -> str:
         return self.name
 
+    @deprecated(
+        "2.8.0dev0",
+        """Use `python_distribution(entry_points={"console_scripts":{...}})` instead of
+        `python_distribution(provides=setup_py().with_binaries(...))`.
+
+        The entry points is specified as a nested dictionary, with a dictionary for each type of
+        entry point, e.g. "console_scripts" or "gui_scripts" etc. Each dictionary maps the entry
+        point name to either a setuptools entry point ("path.to.module:func") or a Pants target
+        address to a pex_binary target.
```

Comment:
Nice! Maybe tweak to this to put it more into terms of how it's relevant to current `.with_binaries` users?

> The entry points field was added as a more generic mechanism than `.with_binaries()`. Whereas `.with_binaries()` would only add `console_scripts` to your generated `setup.py`, you can now add other types of entry points like `gui_scripts`. You can also now add a setuptools entry point like `path.to.module:func`, in addition to still being able to sue a Pants target address to a `pex_binary` target.

Then, in a new paragraph, get into some of the syntactical differences:

> Entry points are specified as a nested dictionary, with a dictionary for each type of entry point like "console_scripts" and "gui_scripts". Each dictionary maps the entry point name to either a setuptools entry point or an address to a pex_binary target.
>
> To migrate, add a dictionary for "console_scripts" with the same entry point name and pex_binary address you were using before. For example: